src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2782,22 +2777,22 @@ Unit* Creature::SelectNearestHostileUnitInAggroRange(bool useLOS) const
     return target;
 }
 
-void Creature::UpdateMovementFlags()
+void Creature::UpdateMovementFlags() // i'm wandering if this method really needs to be called periodically. shouldn't this be done only once, at creation of the creature ? And for transitions (land -> swim, air -> land, etc..), they are usually scriptted therefore the change of movement flag should be done by script and not automatically applied.
```

Comment:
It does some expensive vmap queries each update, maybe it could be throttled a little